impress-org/givewp
# Donor Profiles are now extensible

Resolves #5434 

## Description

This PR introduces Tab registration logic needed to make the Donor Profiles app extensible by addons. Specifically, it introduces the Tabs register, as well as Tab and Route abstracts to help addons easily extend the Donor Profiles app. 

## Affects
In addition to introducing the Tabs register, this PR refactors existing core Tab logic so that all tabs are registered through the Tab register consistently. 

## Pre-review Checklist

## Testing Instructions

1. Add a new Donor Profile block to an empty page
2. Check that all tabs load as expected
